### DISCUSSION TITLE
Not allowed to load local resource

### DISCUSSION THREAD

**Discussion by RamyBouchareb25** [tauri-apps/tauri#11919] [Q&A] [answered]
# Bug: Unable to Load Local Resources in Tauri with React Frontend

## Description
When trying to load image files located on the local filesystem in a Tauri app (using a React frontend), the app fails to render the image. Specifically, using a local file path like `C:\\Users\\ramyb\\Pictures\\mejpg.jpg` in an `<img>` tag results in the following error:

**Error:** `Not allowed to load local resource`

This issue occurs even though the path is valid and accessible on the local machine.

---

## Steps to Reproduce
1. Create a React component with an `<img>` tag.
2. Use a local file path as the `src` value.
3. Launch the Tauri app.
4. Observe the error in the console or failure to render the image.

Example code:
```jsx
const testImage = "C:\\Users\\ramyb\\Pictures\\mejpg.jpg";
return <img src={testImage} alt="Local Test Image" />;
```

---

## Expected Behavior
The image should render properly if the file exists and the path is correct.

---

## Actual Behavior
The image fails to load, and the console shows the error:
```
Not allowed to load local resource
```

---

## Environment
- **OS:** (Windows 11)
- **Tauri Version:** (v2.1.0)
- **React Version:** (v18)
- **Rust Version:** (v1.82.0)
- **Database:** SQLite (using Diesel)

---

### React Component
```javascript
const testImage = "C:\\Users\\ramyb\\Pictures\\mejpg.jpg";
return <img src={testImage} alt="Local Test Image" />;
```

### `lib.rs`
```rust
use core::{establish_connection, load_sqlite_dll};
use diesel::SqliteConnection;
use std::sync::Mutex;

struct AppState {
    db: Mutex<SqliteConnection>,
}

pub fn run() {
    load_sqlite_dll();
    tauri::Builder::default()
        .setup(|app| {
            if cfg!(debug_assertions) {
                app.handle().plugin(
                    tauri_plugin_log::Builder::default()
                        .level(log::LevelFilter::Info)
                        .build(),
                )?;
            }
            Ok(())
        })
        .manage(AppState {
            db: Mutex::new(establish_connection()),
        })
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}

**Comment by FabianLars** [comment, marked as answer]
i think plain file paths are generally not valid img sources. on top of that `file://` urls aren't supported in tauri. Instead there's the `asset:` protocol (because we can't overwrite the `file:` protocol). The exact protocol differs per OS though so you should use convertFileSrc if possible https://v2.tauri.app/reference/javascript/api/namespacecore/#convertfilesrc
